src/earlybird.ml

```diff
@@ -0,0 +1,85 @@
+open Import
+
+module VariableGetClosureCodeLocation = struct
+  let command = "variableGetClosureCodeLocation"
+
+  module Args = struct
+    type t = { handle : int } [@@js]
+  end
+
+  module Result = struct
+    type position = int * int [@@js]
+
+    let position_to_vscode (line, character) =
+      Position.make ~line:(line - 1) ~character:(character - 1)
+
+    type range =
+      { source : string
+      ; pos : position
+      ; end_ : position [@js "end_"]
+      }
+    [@@js]
+
+    let range_to_vscode { pos; end_; _ } =
+      let start = position_to_vscode pos in
+      let end_ = position_to_vscode end_ in
+      Range.makePositions ~start ~end_
+
+    type t = { location : range option } [@@js]
+  end
+end
+
+let debugType = Extension_consts.Debuggers.earlybird
+
+let createDebugAdapterDescriptor ~instance ~session:_ ~executable:_ =
+  let sandbox = Extension_instance.sandbox instance in
+  let command = Sandbox.get_command sandbox "ocamlearlybird" [ "debug" ] in
+  let { Cmd.bin; args } = Cmd.to_spawn command in
```

Comment:
Can we check if the binary is available before trying to run something? In particular, it would be nice to give the user a hint as to which package should be installed.

Reply:
#650 had something to that extent but I removed it when cleaning things up. Earlybird's own VS Code extension didn't even have such check.

I'll give it a shot.

Reply:
The check and error message are now added.